Onboarding views

shaavan opened the pull request:
This PR adds the views for the onboarding wizard.

- The First commit:
    - modifies the Header.qml and TextButton.qml files.
    - It adds options not to wrap text in Header.qml. This is used later in the Introduction.qml file, which needs text in a straight line.
    - It adds options to bold the text and right-align it in the TextButton.qml. This feature is used later to properly align the button’s text on the Developer Options page.
- The Second commit:
    - Adds the Information.qml file
    - This is a modified version of Settings.qml.
    - Instead of a switch button, this allows having text in its place.
    - The text could also double as a hyperlink if the link is provided.
- The Third commit:
    - Adds the first three pages of onboarding wizard:
        - Onboarding01a.qml: Bitcoin core App
        
        
![Screenshot from 2022-06-01 18-58-49](https://user-images.githubusercontent.com/85434418/171422135-e3d95871-0759-4833-88ae-9ba897f8f549.png)

        
        - Onboarding01b.qml: About Options
        
![Screenshot from 2022-06-01 19-00-52](https://user-images.githubusercontent.com/85434418/171422217-dfdb7871-c5b9-4380-8a95-0ba91e427008.png)

        
        - Onboarding01c.qml: Developer Options
        
![Screenshot from 2022-06-01 19-01-24](https://user-images.githubusercontent.com/85434418/171422315-80641ac7-8656-4b56-ba60-55ffe8fa9123.png)

        
    - This also adds the onboarding01.qml file, which works as the connector between the above 3 pages.
    - Adds the AboutOptions.qml and DeveloperOptions.qml file:
        - The AboutOptions.qml file displays the list of options present on the About Options page. And it utilizes the Information.qml file mentioned above to create those sets of options.
        - The DeveloperOptions.qml file creates options on the 01c page.
- The Fourth commit:
    - Adds the onboarding02.qml: Strengthen the bitcoin page
    
![Screenshot from 2022-06-01 19-30-50](https://user-images.githubusercontent.com/85434418/171423024-a868e5c7-5317-4e4a-9f41-b652df991d08.png)

    
    - The node web image is temporarily replaced with the Bitcoin core App icon.
- The Fifth commit:
    - Adds the onboarding03.qml: The block clock page
    
![Screenshot from 2022-06-01 19-09-04](https://user-images.githubusercontent.com/85434418/171422385-057c8648-b977-4b20-8974-1f5f5c561a07.png)
    
    - The block clock image is temporarily replaced with the Bitcoin core App icon.
- The Sixth commit:
    - Adds the onboarding04.qml: Storage Locations
    
![image](https://user-images.githubusercontent.com/85434418/171423182-f03e83f0-a83c-4d06-b5a5-a0359e3b92c2.png)

    
    - This commit also adds StorageLocation.qml component file. This file contains the two options displayed on the page.
- The Seventh commit:
    - Adds the following two pages in the onboarding wizard:
        - onboarding05a.qml: “Storage”
        
![image](https://user-images.githubusercontent.com/85434418/171423209-907679c0-76b2-45f9-ae0d-ed250f9f5928.png)

        
        - onboarding05b.qml: “Storage settings.”
        
 
![image](https://user-images.githubusercontent.com/85434418/171423238-7933f53d-f668-4108-b18e-c9cbf0635002.png)

        
    - The onboarding05.qml page works as the connector between the above two pages.
    - This commit also adds the StorageSettings.qml component page, which contains the list of settings for the 05b.qml page.
- The Eight commit:
    - Adds the last two pages on the onboarding wizard.
        - onboarding06a.qml: “Connection”
        
![image](https://user-images.githubusercontent.com/85434418/171423289-7744bc99-1bfd-42a1-be1c-4b657a4186bc.png)

        
        - onboarding06b.qml: “Connection settings”
        
![image](https://user-images.githubusercontent.com/85434418/171423331-db84e9f8-df02-4a63-9fc0-7eacfdf43a00.png)

        
    - This also adds the onboarding06.qml file, which functions as the connector between the abovementioned pages.
    - This commit updates the ConnectionSettings.qml file to display all the list of settings mentioned in the design file.
- The Ninth commit:
    - Updates the stub.qml file to display the onboarding wizard.


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/124)
[![macOS x86_64](https://img.shields.io/badge/OS-macOS%2C%20x86__64-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/124)
[![macOS arm64_64](https://img.shields.io/badge/OS-macOS%2C%20arm64-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/124)
[![Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/124)

hebasto commented:
Concept ACK. This PR definitely requires designers' reviewing :)

hebasto commented:
Adjust layouts according to https://bitcoindesign.github.io/Bitcoin-Core-App/layout/?

Rspigler commented:
I'm not sure what I am doing wrong here trying to test?
Installed `qml` but still only seeing `bitcoin-qt` (no qml binary).  Running it still results in the old setup.

shaavan commented:
This might be a basic suggestion, but have you installed the QML-specific [dependencies](https://github.com/bitcoin-core/gui-qml/tree/main/src/qml#dependencies)?

I am mentioning this because I recently made a similar mistake :)

In case you have installed the dependencies, it would be best if you could run:

```
./configure --without-bdb --with-qml
```

And share the resultant log part relevant to Qt and QML.

The relevant part should look something like this:

![Screenshot from 2022-06-04 04-27-39](https://user-images.githubusercontent.com/85434418/171964570-9f2c417c-9f25-4206-bee5-842af66727e9.png)

hebasto commented:
Yes, the binary has the same name.

On Sat, 4 Jun 2022 at 00:44, Robert Spigler ***@***.***>
wrote:

> I'm not sure what I am doing wrong here trying to test?
> Installed qml but still only seeing bitcoin-qt (no qml binary). Running
> it still results in the old setup.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin-core/gui-qml/pull/124#issuecomment-1146428070>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH3PXPXGMSORHDT3UV66FY3VNKDEBANCNFSM5XPAR2AA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 

--
Hennadii Stepanov

Rspigler commented:
Ah ok thank you both.  I was looking for the dependencies here: https://github.com/bitcoin-core/gui-qml/blob/main/doc/dependencies.md (which had no change from bitcoin/bitcoin).  Was wondering where the documentation was. Will update in a moment

Rspigler commented:
Notes:

On the 'Info' Page (About; Bitcoin Core is an Open Source Project...):
- The only arrow that leads to another page is 'Developer Options
- I thought I was stuck on that page, until I make the app size bigger - that is when the 'Done' button appeared.
- Storage Limit and Other Option does not do anything (this is probably known, just writing down to make sure)


Strengthen Bitcoin:

- Bitcoin Core runs a full Bitcoin node/,/ which verifies /that/ the rules of the network ....

Block Clock:

- Sometimes it's faster and sometimes /it's/ slower
- The block clock indicates each block on a dial/, which/ represents the current day.

Storage Location:

- I'm assuming that later on it will actually grab the true space available
- 500 GB is large enough for a full node. But how will we determine this for the user. Enough for current storage and 6 months out? A year out? Including indexes?

Storage:
:heavy_check_mark: 

Connection:

- Fast always on/Slow always on have the same description
- For the Desktop Application, what will this page look like?
- Enable Listening, Blocks Only, and Networks should maybe be taken out of Detailed Settings?

Other:
Selecting 'Back' is confusing when in a page's Detailed Settings.  For example, there is Page 1, Page 2, Page 2 Detailed Settings.  Selecting 'Done' returns to Page 2, but selecting 'Back' returns to Page 1.  Perhaps for the Detailed Settings pages there shouldn't be a 'Back' button, only 'Done'?

Possible other config options to add:
Assumevalid
Indexes:

- Blockfilter
- Coinstats
- Tx

Dbcache
Address/change type

Rspigler commented:
I'm also interested in how we will set up the Networks page:

cjdns/i2p/tor/ and/or clear (also incoming/outgoing).

shaavan commented:
Thank you very much, @Rspigler, for your thorough review and analysis of the PR.

Let me try to address your observations as best as I can.

> I thought I was stuck on that page, until I make the app size bigger - that is when the 'Done' button appeared.

That is due to improper layouting. I am working on fixing the layouts according to [design guidelines](https://bitcoindesign.github.io/Bitcoin-Core-App/layout/), which will fix this issue.

> Storage Limit and Other Option does not do anything (this is probably known, just writing down to make sure)

Yes, that's a known limitation. I displayed them as plain text for this PR's initial version to finish up a working prototype of onboarding views. This is intended to fix in the next update of this PR.

> I'm assuming that later on it will actually grab the true space available

Yes. Though this is out of the scope of this PR, this is something we intend to accomplish.

> Fast always on/Slow always on have the same description

Thanks for pointing that out! This was something I missed while writing the description.

> For the Desktop Application, what will this page look like?

We indeed to keep this page the same as displayed through this PR. But it seems like you have some nice ideas to improve this page for the Desktop. If so, I would love to hear them.

> Enable Listening, Blocks Only, and Networks should maybe be taken out of Detailed Settings?

I want to think more about this. Would you please share your reasoning for doing this change?

>  Perhaps for the Detailed Settings pages there shouldn't be a 'Back' button, only 'Done'?

This is something I would definitely agree with. I myself got confused a lot of times while testing this PR.

<details>
<summary> an anecdote of confusion:</summary>

I got to a **detailed setting** page, pressed the **back** button, pressed the **next**, and could not understand why the **done** button was showing up instead of **next**.
</details>

> Possible other config options to add.

Let us give a thought about it.


Thank you very much again for your valuable feedback and thorough analysis. Your review has given me a lot of nice things to work upon :)

Rspigler commented:
Hi @shaavan, thank you for your response!

> We indeed to keep this page the same as displayed through this PR. But it seems like you have some nice ideas to improve this page for the Desktop. If so, I would love to hear them.

The current page has 3 settings:

1. Fast always on (Loads quickly at all times and uses as much cellular data as needed)
2. Slow always on (Loads quickly [sic] at all times and uses as much cellular data as needed)
3. Only when on Wi-Fi (Loads quickly when on wi-fi and pauses when on cellular data)
This works perfectly (IMO) for a cellphone app. However, for the typical binary download or self-compile on a laptop/desktop machine, the first two won't be selected.

Bitcoin/bitcoin has documentation on how to manage traffic for Bitcoin Core: (https://github.com/bitcoin/bitcoin/blob/master/doc/reduce-traffic.md).  These are `maxuploadtarget`, (disable)`listen`, `maxconnections`, and `blocksonly`.  I think the most common options are enable/disable listening and enable blocksonly; these should be the connections settings for the desktop application.  The other settings can be in Detailed Settings.



>>     Enable Listening, Blocks Only, and Networks should maybe be taken out of Detailed Settings?
> I want to think more about this. Would you please share your reasoning for doing this change?

Listening and Blocks Only already discussed above.  For Networks, I think this is also a very important setting, for both privacy reasons and also anti-network partition strength.  Currently, network settings can be very difficult to set up for beginners.

 I can see something like this:

 |         Connection Type                           |                              Outbound                  |                             Inbound                          |
 |-----------------------------|-----------------------------------------------------| --------------------------------------------------------|
 |      Clearnet                |       Disabled/Enabled              |             Disabled/Enabled          |
 |      Proxy                     |       Disabled/Enabled             |             Disabled/Enabled           |
 |     Hidden Service       |      Disabled/Enabled             |              Disabled/Enabled          |


For example, doing nothing and being behind a firewall would have `Clearnet Outbound` `Enabled`, and everything else `Disabled`.  Setting up Listening would enable `Clearnet Inbound`, but only if the firewall allowed it, otherwise there should be a link to port forwarding.

Not all combinations are able to exist, but I think it is a nice way to visualize. (Would also include i2p and cjdns)

hebasto commented:
As I already mentioned during the recent design call, it would be nice to consider the current content of the views as mere placeholders in this PR context, and focus on workflows, transitions, layouts etc.

jarolrod reviewed (commented):
Concept ACK, still reviewing but wanted to talk about the `PageIndicator` which is still at the bottom. It was forgotten in #106 to remove the `PageIndicator` at the bottom. It is no longer to be used there as per discussions had on the design calls and per the design specifications. The screenshots in the PR description crop this out for some reason.

**On PR branch, PageIndicator at bottom**
<img width="912" alt="Screen Shot 2022-06-05 at 9 33 03 PM" src="https://user-images.githubusercontent.com/23396902/172083596-8b5b1c84-0d0f-4cf0-80bc-5c11dbecce79.png">

You should just delete the footer of the `Wizard` control in a new commit, I have done this here: https://github.com/jarolrod/gui-qml/tree/remove-pageindicator-wizard

Feel free to cherry-pick this commit and then rebase the commits included in this PR on top of that.

jarolrod commented:
@Rspigler Thanks for your interest in the project! Right now we're at a stage where we're figuring out processes and in getting an initial app together. There are some elements that will appear out of place, but all of this will be sorted out fairly soon!

shaavan commented:
Updated from e8a6a8db93dd6c3caf5840139770ecd34c2dfeec to fc6f6ea8d940626e1edf35658a5322370d52dc07 ([pr124.01](https://github.com/shaavan/gui-qml/commits/pr124.01) → [pr124.02](https://github.com/shaavan/gui-qml/commits/pr124.02), [diff](https://github.com/shaavan/gui-qml/compare/pr124.01..pr124.02))

Addressed @hebasto, @jarolrod, @Rspigler, and @RandyMcMillan comments

Changes:

- Removed `PageIndicator` bar.
- Added additional property in `Information.qml` to allow text to be editable.
- Done additional changes in the pages to follow the design layouting guide more closely.
- Fix description text for **StorageLocation** page and **ConnectionOptions**.

Rspigler commented:
tACK https://github.com/bitcoin-core/gui-qml/commit/fc6f6ea8d940626e1edf35658a5322370d52dc07
Only think I noticed was that there is still weird behavior between detailed settings/back/done.

shaavan commented:
Updated from fc6f6ea8d940626e1edf35658a5322370d52dc07 to f5a112727405e134796b0f7899c6a1fa57d016fa ([pr124.02](https://github.com/shaavan/gui-qml/commits/pr124.02) -> [pr124.03](https://github.com/shaavan/gui-qml/commits/pr124.02), [diff](https://github.com/shaavan/gui-qml/compare/pr124.02..pr124.03))

Addressed @Rspigler comment

- Removed displaying "< Back" Button for sub-pages.


**Visual Difference:**

|pr124.02|pr124.03|
|------------|------------|
|![image](https://user-images.githubusercontent.com/85434418/173246025-e69803d2-4a55-4582-b833-0f3a037b7acb.png)|![image](https://user-images.githubusercontent.com/85434418/173245982-2f1a52b0-17fd-49c4-a9a4-e37e08015c86.png)|

hebasto commented:
@shaavan 

Maybe rebase this PR on top of the merged #33?

shaavan commented:
Updated from f5a112727405e134796b0f7899c6a1fa57d016fa to 5502d02954323f5f1c199121adbb3be154342454 ([pr124.03](https://github.com/shaavan/gui-qml/commits/pr124.03) -> [pr124.04](https://github.com/shaavan/gui-qml/commits/pr124.04), [diff]())
Addressed @hebasto suggestion

Changes:
- Rebase over the current main.
- Added optional `property` to `Infomation.qml` to use icon instead of descriptive text. Clicking on the button opens the (optional) link.
- Used this new property to display icons in DevelopOptions.qml file.
- Updated "< Back" button from a pure text label to "caret-left + Back text" button.
- Updated "i" button with info.png icon.


Visual differences:

|pr124.03|pr124.04|
|------------|------------|
|![Screenshot from 2022-06-18 00-29-32](https://user-images.githubusercontent.com/85434418/174386649-2c410a47-3317-44d9-9301-339256292df0.png)|![Screenshot from 2022-06-18 00-30-57](https://user-images.githubusercontent.com/85434418/174386670-5abe4399-19b6-423e-a3f3-242e71deb4f4.png)|
|![Screenshot from 2022-06-18 00-29-42](https://user-images.githubusercontent.com/85434418/174386755-00b0663b-396f-4498-867a-a0ad30d7a957.png)|![Screenshot from 2022-06-18 00-31-01](https://user-images.githubusercontent.com/85434418/174386772-bc947d6d-aac0-4b9b-9aa2-772b46d3f127.png)|
|![Screenshot from 2022-06-18 00-29-53](https://user-images.githubusercontent.com/85434418/174386814-e1753cb2-ad2f-4542-acbb-0dd622ffffd8.png)|![Screenshot from 2022-06-18 00-31-08](https://user-images.githubusercontent.com/85434418/174386822-9da7faf6-3214-4f4d-a8cc-47517624333c.png)|

shaavan commented:
Updated from 5502d02954323f5f1c199121adbb3be154342454..a655954d509653f2668546cdb403f2eea0804f2d ([pr124.04](https://github.com/shaavan/gui-qml/commits/pr124.04) -> [pr124.05](https://github.com/shaavan/gui-qml/commits/pr124.05), [diff](https://github.com/shaavan/gui-qml/compare/pr124.04..pr124.05))

Changes:
- Made corrections to remove redundant spacing on the **Developer options** page.

Visual Changes:

|pr124.04|pr124.05|
|------------|-----------|
|![image](https://user-images.githubusercontent.com/85434418/174388490-0a37a5f5-ed9d-4d8e-b989-5f6ca1571e3d.png)|![image](https://user-images.githubusercontent.com/85434418/174388609-b6c6dd60-619b-4180-8462-577f2d0f245a.png)|

Rspigler commented:
tACK a655954d509653f2668546cdb403f2eea0804f2d
Recent changes are implemented

hebasto commented:
@shaavan Please rebase this PR to be able to download Apple M1 binaries for testing.

shaavan commented:
Updated from a655954d509653f2668546cdb403f2eea0804f2d to 11dcbfdee5a174adff60444627c9dd8244d0bc20 ([pr124.05](https://github.com/shaavan/gui-qml/commits/pr124.05) -> [pr124.06](https://github.com/shaavan/gui-qml/commits/pr124.06), [diff](https://github.com/shaavan/gui-qml/compare/pr124.05..pr124.06))

Addressed @hebasto comment.

Changes:
- Rebased the PR over the current master.

hebasto commented:
I've added macOS arm64 binary download link to the PR description.

mouxdesign commented:
Did a test using the windows binary today on my pc. I placed Christophs designs next to the actual screenshots to do the analysis.
Hope this is helpful!
![Group 2367](https://user-images.githubusercontent.com/90313394/177395625-ab95de52-d1a2-4e86-af5e-beb69b828083.jpg)

mouxdesign commented:
Adding in here:

-  Specs of the monitor where the gui was opened
-  The windows architecture.
- Video of how the Gui appears

![Pixels](https://user-images.githubusercontent.com/90313394/177750687-bdbce7cc-bc05-48cd-9827-20306ac79852.jpg)
![System type](https://user-images.githubusercontent.com/90313394/177750707-3506c789-fe85-4bfb-b991-704b05e15a33.jpg)

[screen-capture.webm](https://user-images.githubusercontent.com/90313394/177750939-08f23d67-03ba-4502-ad33-75504a225110.webm)

GBKS commented:
@mouxdesign this looks quite different. Do you maybe have some operating systems for larger fonts or anything like that enabled?

Rspigler commented:
I apologize for not testing carefully enough. I have all the same issues that @mouxdesign has as well.

I am on Fedora 35

GBKS commented:
My guess is that declaring font sizes via `pixelSize` instead of `pointSize` will fix this.

shaavan commented:
Updated from 11dcbfdee5a174adff60444627c9dd8244d0bc20 to 1c64135a68af45455685f6f00904a20875c56e6b ([pr124.06](https://github.com/shaavan/gui-qml/commits/pr124.06) -> [pr124.07](https://github.com/shaavan/gui-qml/commits/pr124.07), [diff](https://github.com/shaavan/gui-qml/compare/pr124.06..pr124.07))

Addressed @GBKS suggestion.

Changes:
- Replaced `pointSize` with `pixelSize`.

There is a noticeable visual change when using the pixelSize. 

**Screenshot:**

|pr124.06|pr124.07|
|---------|----|
|![Screenshot from 2022-07-15 21-50-04](https://user-images.githubusercontent.com/85434418/179267693-e2fefbb2-b59c-43ea-9681-d3bfe2c4cca2.png)|![Screenshot from 2022-07-15 21-51-05](https://user-images.githubusercontent.com/85434418/179267628-b2def07f-b091-48e5-b422-86047307e555.png)|

The design looks much cleaner now after the update.


There are a few more suggestions I would like to address before this PR gets merged. However, currently, I am pushing this single change so that the reviewers can test locally if they can observe the behavior getting corrected on their machines. 

A gentle ping @mouxdesign and @Rspigler.

Thanks, @GBKS, for your suggestions, and thanks, @hebasto, for the ping!

mouxdesign commented:
Super! Great work @shaavan. Could you send the link to binary again? I'll be happy to test it again. Will be amazing to see the gui in in the way that it was designed.

shaavan commented:
Thanks, @mouxdesign! The binary link in the PR's description shall update to the latest version as soon as the CI completes.

Here's the link: https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/124

However, you should wait for checks to complete before downloading using this link.

@hebasto
I am not sure why some CI checks failed. Would you please help in understanding the reason for it? Thanks!

hebasto commented:
> @hebasto
> I am not sure why some CI checks failed. Would you please help in understanding the reason for it? Thanks!

@shaavan 
Suggesting to rebase.

shaavan commented:
Updated from 1c64135a68af45455685f6f00904a20875c56e6b to 802e6826fca6536fa91ad25a96f9cb33f31f6506 ([pr124.07](https://github.com/shaavan/gui-qml/commits/pr124.07) -> [pr124.08](https://github.com/shaavan/gui-qml/commits/pr124.08), [diff](https://github.com/shaavan/gui-qml/compare/pr124.07..pr124.08))

Addressed @hebasto suggestion.

- Rebased PR over current master, to prevent CI failures.

hebasto commented:
@mouxdesign @GBKS 

Since all of the CI tests are green, binaries for testing are available for downloading via the links at the bottom of the PR description as usual :tiger2:

shaavan commented:
Updated from 802e6826fca6536fa91ad25a96f9cb33f31f6506 to 5118be447a705578eb90cf39a0971ddfc53f8d74 ([pr124.08](https://github.com/shaavan/gui-qml/commits/pr124.08) -> [pr124.09](https://github.com/shaavan/gui-qml/commits/pr124.09), [diff](https://github.com/shaavan/gui-qml/compare/pr124.08..pr124.09))

Addessed @GBKS suggestions.

Changes:
- Moved navbar code from `Wizard.qml` to each page separately.
- Fixed Layouting issue by removing redundant right margin and top padding from `Settings.qml` and `Information.qml`.
- Removed the **Done** button on the **About Page** and **Developer options page** and added the `< Back` button on them.
- Moved the **Done** button from the bottom center of the page to the right of the navbar in the **Storage Settings** and **Connection Settings** pages.
- Decreased the maximum content width from `600` -> `450` in **Connection Settings** and **Storage Settings** page.
- Added a missing divider line between Connection Settings page.
- Decreased the minimum app window size to `655` pixels.

Some Minor Changes:
- Fixed Caret-left size for the `< Back` button according to the Design.

Here are the screenshots for all the visual differences after this update:

|pr124.08|pr124.09|
|------------|-----------|
|![Screenshot from 2022-07-17 01-12-24](https://user-images.githubusercontent.com/85434418/179369826-2da59309-7869-4f19-be38-ca40b05e3b53.png)|![Screenshot from 2022-07-17 00-55-07](https://user-images.githubusercontent.com/85434418/179369814-2c9fe453-4474-4623-808b-fbd48b08c93a.png)|
|![Screenshot from 2022-07-17 01-12-29](https://user-images.githubusercontent.com/85434418/179369865-b1e92936-a53e-4406-a6ec-8bdd525c7363.png)|![Screenshot from 2022-07-17 00-55-13](https://user-images.githubusercontent.com/85434418/179369872-3bc4b46f-d7eb-4c0a-bda5-823f6ac94083.png)|
|![Screenshot from 2022-07-17 01-12-40](https://user-images.githubusercontent.com/85434418/179369877-4522aff8-0e0f-4c46-b221-debffb123ec8.png)|![Screenshot from 2022-07-17 00-55-30](https://user-images.githubusercontent.com/85434418/179369882-b2eaae43-ee66-4bb0-8dd9-326c07ac0ce6.png)|
|![Screenshot from 2022-07-17 01-12-49](https://user-images.githubusercontent.com/85434418/179369886-48b215af-2a8f-4fd6-8800-bd3129b9c2b2.png)|![Screenshot from 2022-07-17 00-55-38](https://user-images.githubusercontent.com/85434418/179369889-03357596-b75a-4c04-b5c9-1a106f21c468.png)|


First of all, thank you very much, @GBKS, for all your valuable suggestions. All these suggestions helped improve the PR a lot. 
I would also like to address the suggestion I did not include in this PR.

1. Replace the Bitcoin app icon with appropriate images on the Strengthen Bitcoin (page 02) and The Block Clock (page 03).
    - Since there is a need for discussion on whether these images should be used statically or should be animated, I decided to keep a hold on implementing this change.
2. Circle the "i" icon in the black when the light theme on Start (01a) page
    - Doing this would require the "white" version of the "I" icon; that is, the info.svg with the color="white".

|Current Image|Required Image|
|-------------------|---------------------|
|![image](https://user-images.githubusercontent.com/85434418/179369623-8e40ff83-1263-48b8-9136-5824d3ad9415.png)|![image](https://user-images.githubusercontent.com/85434418/179369651-f4889b85-b69d-4d87-96b8-9323e0812647.png)|

Since I wanted to be sure of the correct method of importing svg, and png format in resources, I decided to keep this change on hold.

Again thank you very much for all your suggestions!

A friendly ping to my fellow contributors: @hebasto @jarolrod @mouxdesign @Rspigler @RandyMcMillan

Rspigler commented:
Will test shortly

Rspigler commented:
tACK commit 5118be447a705578eb90cf39a0971ddfc53f8d74

I can confirm that the text size changes from pr124.06->pr124.07 has taken place
I can confirm the visual changes mentioned in comment https://github.com/bitcoin-core/gui-qml/pull/124#issuecomment-1186277484

GBKS commented:
Great update, @shaavan.

Just tested, and noticed a few more things we can address, whether in this PR or elsewhere:
- Input field types, states, validation and min/max values (I can enter "Pizza" for daily upload limit)
- Caret icon for list items (instead of the ">" character)
- Animate the toggle circle when switching on/off (super minor nice-to-have)

Was also wondering if we need more, and/or more visual, hover/selected states. When hovering a web link on "About", for example, I see the text select cursor. Should probably be the hand cursor, and we could change the text color (to orange) for clearer feedback.

jarolrod commented on src/qml/controls/Setting.qml:
```diff
@@ -26,6 +26,7 @@ Control {
             }
             OptionSwitch {
                 Layout.alignment: Qt.AlignRight
+                Layout.rightMargin: -12
```
It's not clear to me why we are hardcoding a `rightMargin` value of `12` here. This doesn't seem like the correct way to do this, what is the design spec that inspired this change?

shaavan replied:
The `rightMargin` was explicitly set to correctly position the `OptionsSwitch`, to be precisely above the right end of the divider line.

|Before|After|
|---------|------|
|![Screenshot from 2022-08-01 18-07-52](https://user-images.githubusercontent.com/85434418/182149960-f57f0266-7806-4733-88c8-cccb466e82d9.png)|![Screenshot from 2022-08-01 18-08-04](https://user-images.githubusercontent.com/85434418/182149995-bafc6d03-e1be-4a14-baeb-70d68e215e53.png)|

I experimented with other _conventional_ methods of aligning elements using `Layouts` and `Anchors` before going with hardcoded margin. Still, none of them worked to position the Switch over the divider properly.

jarolrod commented on src/qml/controls/Information.qml:
```diff
@@ -0,0 +1,88 @@
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+Control {
```
As the description of a17b84ec675c8ab27a560d9d301c8f8d96d90443 states, "This is an object similar to Setting.qml". I think that instead of having this control we can just refactor `Setting.qml` to load the desired component in place of the currently hardcoded switch. This adds flexibility to the setting control to fit in with all the use cases required of it within options handling without introducing a new component here, and other components when a new component is desired in the switch's position.

shaavan replied:
This would be the ideal way to move forward, and this was something I tried to do while beginning to write this PR.

However, the crucial problem I faced while going with this idea at the beginning was that QML is a declarative language and hence doesn't support conditional statements that are more complicated than a ternary operator.
So it was quite hard to figure out a way to make `Settings.qml` switch the type of _description_ based on the option type without using conditional statements. So I went forward by creating `Information.qml` from scratch and, one by one, incorporated changes to work with all the option types. 

But now, since the `Information.qml` has been laid out, and almost all the permutations of various options have been accounted for, we can think of ways to merge these two files effectively.

jarolrod commented on src/qml/components/AboutOptions.qml:
```diff
@@ -0,0 +1,65 @@
+    Information {
+        Layout.fillWidth: true
+        header:  qsTr("Website")
+        description: qsTr("bitcoincore.org >")
```
The descriptions here should not be translatable.

shaavan replied:
Addressed in #140
Thanks!

jarolrod commented on src/qml/controls/Header.qml:
```diff
@@ -27,11 +28,11 @@ Control {
             font.family: "Inter"
             font.styleName: root.bold ? "Semi Bold" : "Regular"
-            font.pointSize: root.headerSize
+            font.pixelSize: root.headerSize
```
in 8763690db1af6c3956c4f3fc60a30b445db5a01f, you didn't update the PR description to state that we are now using `pixelSize` instead of `pointSize` when you updated the PR. This is a pretty important change to document. I would suggest extracting out this part into it's own commit (adding a nice description to document the historical reason):

qml: use font pixelSize instead of pointSize
```diff
diff --git a/src/qml/controls/ContinueButton.qml b/src/qml/controls/ContinueButton.qml
index 2b98f8baf..d478ba45e 100644
--- a/src/qml/controls/ContinueButton.qml
+++ b/src/qml/controls/ContinueButton.qml
@@ -8,7 +8,7 @@ import QtQuick.Controls 2.15
 Button {
     font.family: "Inter"
     font.styleName: "Semi Bold"
-    font.pointSize: 18
+    font.pixelSize: 18
     contentItem: Text {
         text: parent.text
         font: parent.font
diff --git a/src/qml/controls/Header.qml b/src/qml/controls/Header.qml
index 13d42e1c4..59fb01671 100644
--- a/src/qml/controls/Header.qml
+++ b/src/qml/controls/Header.qml
@@ -27,7 +27,7 @@ Control {
             topPadding: root.headerMargin
             font.family: "Inter"
             font.styleName: root.bold ? "Semi Bold" : "Regular"
-            font.pointSize: root.headerSize
+            font.pixelSize: root.headerSize
             color: Theme.color.neutral9
             text: root.header
             horizontalAlignment: center ? Text.AlignHCenter : Text.AlignLeft
@@ -42,7 +42,7 @@ Control {
                 topPadding: root.descriptionMargin
                 font.family: "Inter"
                 font.styleName: "Regular"
-                font.pointSize: root.descriptionSize
+                font.pixelSize: root.descriptionSize
                 color: Theme.color.neutral8
                 text: root.description
                 horizontalAlignment: root.center ? Text.AlignHCenter : Text.AlignLeft
diff --git a/src/qml/controls/TextButton.qml b/src/qml/controls/TextButton.qml
index 1063a275b..2dce44670 100644
--- a/src/qml/controls/TextButton.qml
+++ b/src/qml/controls/TextButton.qml
@@ -11,7 +11,7 @@ Button {
     property string textColor: Theme.color.neutral9
     font.family: "Inter"
     font.styleName: "Semi Bold"
-    font.pointSize: root.textSize
+    font.pixelSize: root.textSize
     contentItem: Text {
         text: root.text
         font: root.font

```

jarolrod commented on src/qml/controls/Wizard.qml:
```diff
@@ -11,18 +11,6 @@ Page {
     property var views
     property alias finished: swipeView.finished
     background: null
-    header: RowLayout {
```
in 79e959c12f3c7ed3e90619f4cf969325f2c46a0b

a better title is `qml: remove page indicator and navbar from Wizard Control`

jarolrod commented on src/qml/controls/Header.qml:
```diff
@@ -19,6 +19,7 @@ Control {
     property string subtext
     property int subtextMargin
     property int subtextSize: 15
+    property bool wrap: true
```
you could make this it's own commit:

qml: customizable wrap mode in Header control
``` diff
diff --git a/src/qml/controls/Header.qml b/src/qml/controls/Header.qml
index 59fb01671..c7adcf309 100644
--- a/src/qml/controls/Header.qml
+++ b/src/qml/controls/Header.qml
@@ -19,6 +19,7 @@ Control {
     property string subtext
     property int subtextMargin
     property int subtextSize: 15
+    property bool wrap: true
     contentItem: ColumnLayout {
         spacing: 0
         Label {
@@ -31,7 +32,7 @@ Control {
             color: Theme.color.neutral9
             text: root.header
             horizontalAlignment: center ? Text.AlignHCenter : Text.AlignLeft
-            wrapMode: Text.WordWrap
+            wrapMode: wrap ? Text.WordWrap : Text.NoWrap
         }
         Loader {
             Layout.fillWidth: true
@@ -46,7 +47,7 @@ Control {
                 color: Theme.color.neutral8
                 text: root.description
                 horizontalAlignment: root.center ? Text.AlignHCenter : Text.AlignLeft
-                wrapMode: Text.WordWrap
+                wrapMode: wrap ? Text.WordWrap : Text.NoWrap
             }
         }
         Loader {
@@ -62,7 +63,7 @@ Control {
                 color: Theme.color.neutral9
                 text: root.subtext
                 horizontalAlignment: root.center ? Text.AlignHCenter : Text.AlignLeft
-                wrapMode: Text.WordWrap
+                wrapMode: wrap ? Text.WordWrap : Text.NoWrap
             }
         }
     }
```

jarolrod commented on src/qml/controls/TextButton.qml:
```diff
@@ -9,14 +9,16 @@ Button {
     id: root
     property int textSize: 18
     property string textColor: Theme.color.neutral9
+    property bool bold: true
```
you could make this it's own commit, also maybe switch the logic from defining `rightalign` to `center`?:

qml: make text boldness and alignment customizable in TextButton control
```diff

diff --git a/src/qml/controls/TextButton.qml b/src/qml/controls/TextButton.qml
index 2dce44670..4f508970d 100644
--- a/src/qml/controls/TextButton.qml
+++ b/src/qml/controls/TextButton.qml
@@ -10,13 +10,15 @@ Button {
     property int textSize: 18
     property string textColor: Theme.color.neutral9
     font.family: "Inter"
-    font.styleName: "Semi Bold"
+    font.styleName: bold ? "Semi Bold" : "Regular"
     font.pixelSize: root.textSize
+    property bool bold: true
+    property bool center: true
     contentItem: Text {
         text: root.text
         font: root.font
         color: root.textColor
-        horizontalAlignment: Text.AlignHCenter
+        horizontalAlignment: center ? Text.AlignHCenter : Text.AlignRight
         verticalAlignment: Text.AlignVCenter
     }
     background: null
```

GBKS commented:
ACK 5118be447a705578eb90cf39a0971ddfc53f8d74

jarolrod commented:
ACK https://github.com/bitcoin-core/gui-qml/commit/5118be447a705578eb90cf39a0971ddfc53f8d74

Will pick up feedback in follow-ups

hebasto commented:
Observing in terminal:
```
Unsupported modifier, resource creation failed.
XXX: resource creation failed
```

shaavan commented:
@jarolrod 

> Will pick up feedback in follow-ups

Great! Let me know which one you pick first, and I shall get doing the rest.

shaavan commented:
@hebasto 

https://github.com/bitcoin-core/gui-qml/pull/124#issuecomment-1197117450

I didn't observe this message while building/running the views.

It would greatly help if you could give details on precisely when you observed this statement: when starting the GUI or after clicking on some action in the views.

hebasto commented:
> [#124 (comment)](https://github.com/bitcoin-core/gui-qml/pull/124#issuecomment-1197117450)
> 
> I didn't observe this message while building/running the views.
> 
> It would greatly help if you could give details on precisely when you observed this statement: when starting the GUI or after clicking on some action in the views.

It seems like a Wayland-specific issue.

shaavan commented:
> It seems like a Wayland-specific issue.

So is this something we can work on fixing, or is it a bug intrinsic to Wayland?

hebasto commented:
> > It seems like a Wayland-specific issue.
> 
> So is this something we can work on fixing, or is it a bug intrinsic to Wayland?

IDK :)

shaavan commented on src/qml/controls/Information.qml:
```diff
@@ -0,0 +1,88 @@
+                Layout.alignment: Qt.AlignRight | Qt.AlignHCenter
+                active: root.description.length > 0
+                visible: active
+                sourceComponent: TextEdit {
```
The `TextEdit` Component is needed for the options like `Storage Limit` or `Connection Limit`, which required the values to be editable.

Getting rid of the `readonly` property would mean creating separate loaders for editable and non-editable text.

I would like to hear your take on this and would be glad to know a more efficient way to deal with this problem.

mouxdesign commented:
Just downloaded the binary and I'm happy to say that it looks great! I don't have the font sizing issues I had the first time. In the image below row 2, you can see the present version, row 1 was the previous version. Well done @shaavan <br/>
![Group 2367](https://user-images.githubusercontent.com/90313394/182537417-5c4cf1d2-5bda-442d-b013-cbda3cc9d645.png)

Rspigler commented:
@mouxdesign Do you mean column 2/3? The images below row 2 still show many errors

Rspigler commented:
I tested this again, and my system shows column 2

GBKS commented:
@Rspigler I believe column 2 is the newer and correct version - so should be all good. Is that right, @mouxdesign?

mouxdesign commented:
100% correct as @GBKS mentioned the second column is the newer version. I see in my initial message I mentioned a row, so understand which the confusion might have come from. The third column are Christoph's original designs.
